source/common/modules/markdown-editor/plugins/render-math.ts

```diff
@@ -15,32 +15,29 @@
 import { getBlockMathRE, getInlineMathRenderRE } from '@common/regular-expressions'
 import CodeMirror from 'codemirror'
 import katex from 'katex'
-
+import { IpcRenderer } from 'electron'
 import 'katex/contrib/mhchem' // modify katex module
+import { LatexCommand } from '@providers/latex-commands'
+
+const ipcRenderer: IpcRenderer = (window as any).ipc
 
 const multilineMathRE = getBlockMathRE()
 const commands = (CodeMirror.commands as any)
-commands.markdownRenderMath = function (cm: CodeMirror.Editor) {
+commands.markdownRenderMath = async function (cm: CodeMirror.Editor) {
```

Comment:
Please no async commands. This way we have no way of catching errors since CM doesn't handle these. Instead, it's better to outsource async code to an external function that you could call from time to time and which updates a variable instead.